File: ulyp-ui/src/main/java/com/ulyp/ui/AggregationStrategy.java
package com.ulyp.ui;

import com.ulyp.core.CallRecordDatabase;
import com.ulyp.core.MethodInfoDatabase;
import com.ulyp.core.TypeInfoDatabase;

public interface AggregationStrategy {

    CallRecordTreeTabId getId(CallRecordTreeChunk chunk);

    CallRecordDatabase buildDatabase(MethodInfoDatabase methodInfoDatabase, TypeInfoDatabase typeInfoDatabase);
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/AggregationStrategy.kt
package com.ulyp.ui

import com.ulyp.core.CallRecordDatabase
import com.ulyp.core.MethodInfoDatabase
import com.ulyp.core.TypeInfoDatabase

interface AggregationStrategy {

    fun getId(chunk: CallRecordTreeChunk): CallRecordTreeTabId

    fun buildDatabase(methodInfoDatabase: MethodInfoDatabase?, typeInfoDatabase: TypeInfoDatabase?): CallRecordDatabase
}